WordPress/Sniffs/WhiteSpace/ControlStructureSpacingSniff.php

```diff
@@ -344,6 +344,7 @@ public function process_token( $stackPtr ) {
 				}
 
 				if ( \T_WHITESPACE !== $this->tokens[ ( $parenthesisCloser + 1 ) ]['code']
+					&& T_COLON !== $this->tokens[ ( $parenthesisCloser + 1 ) ]['code']
```

Comment:
I'm trying to think of code samples on which this will break, but it's late and my brain is shutting down.

Maybe - to be on the safe side -, this condition should be combined with a check that the `$stackPtr` which triggered the sniff is either a `T_FUNCTION` or a `T_CLOSURE` token ?

Reply:
I would agree, but I'm not sure what `$stackPtr` would point to at this moment of the process.

I considered trying to find the next `{`, and then see if the scope_condition was a `T_FUNCTION`, or other similar things to find out of the `:` is indeed using somewhere inside `function ... {`.

I'll see if I can review what `$stackPtr` should contain.

Reply:
Finally got my head around the approach I was trying to take. See 07bc4ba.

Reply:
Sorry, I was in a meeting. I think the code can actually be simpler:
You already have the `$parenthesisCloser` and - if I remember correctly - parenthesis which belong to a function declaration will have a `parenthesis_owner` set in the token array, so you could look at that and check if that is a `T_FUNCTION` or `T_CLOSURE`.

Reply:
Awesome, thanks - https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/1421/commits/798be40c739796b15f648dc28c9f2ce7367b1f59.

Reply:
And https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/1421/commits/d2685e9453b77b21f3d7e970a0507cd6ecff2707 updates again to support closures.